src/components/__tests__/HeadingBackground.cy.js

```diff
@@ -1,48 +1,138 @@
+
+// variables
+const sectionImageWidthMobile = 100;
+const sectionTextWidthMobile = 100;
 
 describe('<HeadingBackground>', () => {
-  beforeEach(() => {
-    cy.mount(HeadingBackground, {
-      props: {
-        title,
-      },
-    });
+  it('has translation for all strings', () => {
+    cy.testLanguageStringsInContext(
+      ['highlightCommunity', 'imageAltText', 'titleCommunity'],
+      'index.headingBackground',
+      i18n,
+    );
   });
 
-  it('renders title with correct styling', () => {
-    cy.window().then(() => {
-      cy.dataCy('heading-wrapper').should('be.visible');
-      cy.dataCy('heading')
-        .should('be.visible')
-        .and('contain', title)
-        .then(($title) => {
-          expect($title.text()).to.equal(title);
-        });
-      cy.dataCy('heading')
-        .should('have.css', 'font-size', '24px')
-        .and('have.css', 'font-weight', '700');
+  context('desktop', () => {
+    beforeEach(() => {
+      cy.mount(HeadingBackground, {
+        props: {},
+      });
     });
+
+    coreTests();
+
+    it('renders sections side-by-side', () => {
+      cy.viewport('macbook-16');
+      cy.testElementsSideBySide(selectorSectionText, selectorSectionImage);
+    });
+
+    it('renders image', () => {
+      cy.dataCy(selectorHeadingBackgroundImage).then((element) => {
+        cy.wrap(element).matchImageSnapshot({
+          failureThreshold: 0.1,
+          failureThresholdType: 'percent',
+          timeout: 5000,
+          customDiffConfig: { threshold: 0.5 },
+          screenshotsFolder: 'test/cypress/snapshots',
+          retries: 2,
+          name: `heading-background-image-${Cypress.currentTest.titlePath[0]}`,
+        });
+      });
+    })
   });
 
-  it('renders svg background', () => {
-    cy.window().then(() => {
-      cy.dataCy('svg').should('be.visible');
+  context('mobile', () => {
+    beforeEach(() => {
+      cy.mount(HeadingBackground, {
+        props: {},
+      });
       cy.viewport('iphone-6');
-      cy.dataCy('svg')
-        .should('be.visible')
-        .then(($element) => {
-          expect($element.width()).to.be.closeTo(375, 0.5);
-          expect($element.height()).to.be.closeTo(75, 1);
-        });
-      cy.viewport('macbook-13');
-      cy.dataCy('svg')
-        .should('be.visible')
-        .then(($element) => {
-          expect($element.width()).to.be.closeTo(1280, 0.5);
-          expect($element.height()).to.be.closeTo(256, 5);
-        });
     });
+
+    coreTests();
+
+    it('renders sections stacked', () => {
+      cy.testElementPercentageWidth(
+        cy.dataCy(selectorSectionImage),
+        sectionImageWidthMobile,
+      );
+      cy.testElementPercentageWidth(
+        cy.dataCy(selectorSectionText),
+        sectionTextWidthMobile,
+      );
+    });
+
+    it('renders image', () => {
+      cy.dataCy(selectorHeadingBackgroundImage).then((element) => {
+        cy.wrap(element).matchImageSnapshot({
```

Comment:
This func should be Cypress command because settings repeat across all src code.

Reply:
Refactored into a Cypress command.